src/habitat.cr

```diff
@@ -43,7 +45,7 @@ class Habitat
     include Habitat::TempConfig
     include Habitat::SettingsHelpers
 
-    HABITAT_SETTINGS = [] of Crystal::Macros::TypeDeclaration
+    HABITAT_SETTINGS = [] of NamedTuple(type: Crystal::Macros::TypeDeclaration, example: T)
```

Comment:
I think this is the biggest flaw in my implementation: I don't know how to enforce that `example` and the `type` declared have the same type 😞

Reply:
You're right that it isn't available in macros. Macros are more limited in what they can do. In fact the type declaration in this constant doesn't really affect anything since it is only used in a macro. To show that the type is not enforced, it is a common idiom to use `Nil` instead.

`HABITAT_SETTINGS = [] of Nil`. It's a bit weird, but I believe it will work :)